docs/source/configuration.rst
.. _configuration:

==========================
Configuration and Defaults
==========================

Contents:

.. toctree::
    :maxdepth: 2

    ./configuration/configuration-file
    ./configuration/configuration-country-reporting
    ./configuration/configuration-asterisk
    ./configuration/configuration-freeswitch
    ./configuration/configuration-reset-data
    ./celery/celery-configuration
    ./configuration/configuration-acl-control


docs/source/configuration-file.rst
.. _configuration:

Configuration
=============

Some of the more important parts of the configuration module for the cdr_stats,
``settings.py``, are explained below.

``APPLICATION_DIR`` now contains the full path of the project folder and can be used elsewhere
in the ``settings.py`` module so that the project may be moved around the system without having to
worry about changing any hard-coded paths. ::

    import os.path
    APPLICATION_DIR = os.path.dirname(globals()['__file__'])

Turns on debug mode allowing the browser user to see project settings and temporary variables. ::

    DEBUG = True

Sends all errors from the production server to the admin's email address. ::

    ADMINS = ( ('xyz', 'xyz@abc.com') )


Sets up the options required for Django to connect to your database engine::

    DATABASES = {
        'default': {
            # Add 'postgresql_psycopg2','postgresql','mysql','sqlite3','oracle'
            'ENGINE': 'django.db.backends.postgresql_psycopg2',
            'NAME': 'DATABASENAME',
            'USER': 'DB_USERNAME',
            'PASSWORD': 'DB_PASSWORD',
            'HOST': 'DB_HOSTNAME',
            'PORT': 'DB_PORT',
            'OPTIONS': {
                #Needed on Mysql
                # 'init_command': 'SET storage_engine=INNODB',
                #Postgresql Autocommit
                'autocommit': True,
            }
        }
    }


Sets up the options to connect to MongoDB Server, this server and database will be used to store the analytic data.
There is usually no need to change these settings, unless if your MongoDB server is on a remote server, or if
different names are required for the collections::

    #MONGODB
    #=======
    MONGO_CDRSTATS = {
        'DB_NAME': 'cdr-stats',
        'HOST': 'localhost',
        'PORT': 27017,
        'CDR_COMMON': 'cdr_common',
        'DAILY_ANALYTIC': 'daily_analytic',
        'MONTHLY_ANALYTIC': 'monthly_analytic',
        'CONC_CALL': 'concurrent_call',
        'CONC_CALL_AGG': 'concurrent_call_aggregate'
    }


Tells Django where to find your media files such as images that the ``HTML
templates`` might use. ::

    MEDIA_ROOT = os.path.join(APPLICATION_DIR, 'static')

    ROOT_URLCONF = 'urls'


Tells Django to start finding URL matches at in the ``urls.py`` module in the ``cdr_stats`` project folder. ::

      TEMPLATE_DIRS = ( os.path.join(APPLICATION_DIR, 'templates'), )


Tells Django where to find the HTML template files::

    INSTALLED_APPS = (
        'django.contrib.auth',
        'django.contrib.contenttypes',
        'django.contrib.sessions',
        'django.contrib.sites',
        'django.contrib.admin',
        ...
        'cdr',
        'cdr_alert',
        ...
    )

Tells Django which applications (custom and external) to use in the project.
The custom applications, ``cdr`` etc. are stored in the project folder along with
these custom applications.


.. _configuration-country-reporting:

Country Reporting
-----------------

CDR-Stats is able to identify the destination country of the call. This is a
useful fraud prevention measure, so that calls to unexpected destinations
are immediately apparent. Places that should not be called should be added
in the Blacklist in the admin section so that these destinations are
highlighted in the call data records.

However, in order to get accurate reporting, the call detail records have to
be in international format, e.g. in the USA, this means 11 digit numbers,
beginning with a 1, and for other countries, the numbers called should be
prefixed with the international dial code.

There is a facility for manipulating the dialled digits reported in the call
detail records, as well as identifying calls as internal calls. This is done
in the "general" section of /usr/share/cdr-stats/settings_local.py.

PREFIX_LIMIT_MIN & PREFIX_LIMIT_MAX are used to determine how many digits are used to match against the dialcode prefix database, e.g

* **PREFIX_LIMIT_MIN = 2**
* **PREFIX_LIMIT_MAX = 5**

If a phone number has less digits  than PN_MIN_DIGITS it will be considered an extension::

* **PN_MIN_DIGITS = 6**
* **PN_MAX_DIGITS = 9**

If a phone number has more digits than PHONENUMBER_DIGITS_MIN but less than PHONE_DIGITS_MAX then the phone number will be considered as local or national call and the LOCAL_DIALCODE will be added.

* **LOCAL_DIALCODE = 1**

Set the dialcode of your country (44 for UK, 1 for US)

* **PREFIX_TO_IGNORE = "+,0,00,000,0000,00000,011,55555,99999"**

List of prefixes to ignore, these prefixes are removed from the phone number prior to analysis. In cases where
customers dial 9 for an outside line, 9, 90 or 900 may need to be removed as well to ensure accurate reporting.


Examples
~~~~~~~~

So for the USA, to cope with 10 or 11 digit dialling, PN_MAX_DIGITS would be set to 10, and LOCAL_DIALCODE set to 1. Thus 10 digit numbers would have a 1 added, but 11 digit numbers are left untouched.

In the UK, the number of significant digits is either 9 or 10 after the "0" trunk code. So to ensure that all UK numbers had 44 prefixed to them and the single leading 0 removed, the prefixes to ignore would include 0, the PN_MAX_DIGITS would be set to 10, and the LOCAL_DIALCODE would be 44.

In Spain, where there is no "0" trunk code, and the length of all numbers is 9, then the PN_MAX_DIGITS  would be set to 9, and the LOCAL_DIALCODE set to 34.

NB: After changing this file, then both celery and apache should be restarted.


.. _resetting-data:

Resetting Data
--------------

Sometimes, some experimentation is required to get the optimum settings for country reporting, to achieve this the data is removed
from MongoDB and re-imported from the Asterisk MySQL database.

1. Stop Celery

2. Type mongo to enter the MongoDB database then apply the following commands::

    mongo
    use cdr-stats;
    db.monthly_analytic.remove({});
    db.daily_analytic.remove({});
    db.aggregate_world_report.remove({});
    db.aggregate_result_cdr_view.remove({});
    db.aggregate_hourly_country_report.remove({});
    db.cdr_common.remove({});

    CTRL-D exits the console.

3. Flag the CDR records for reimport
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

3a. With Asterisk and Mysql ::

    Go to the CDR database in Asterisk and change the field 'import_cdr' to 0:

    Enter the MySQL console with the following command, changing the credentials and database name
    to suit your installation:
    mysql -uasteriskuser -pamp109 asteriskcdrdb

    update cdr SET import_cdr = 0;

    CTRL-C exits the MySQL


3b. With FreeSWITCH and MongoDB::

    Go to the CDR FreeSWITCH MongoDB database, update all the records by setting the 'import_cdr' field to 0.

    mongo
    use freeswitch_cdr;
    db.cdr.update({"import_cdr" : 1}, { $set : {"import_cdr" : 0}}, { multi: true });

4. Now start Celery.
5. Wait while the CDR are re-imported.


.. _configuration-asterisk:

Import configuration for Asterisk
---------------------------------


The asterisk settings may be as follows::

    #list of CDR Backends to import
    CDR_BACKEND = {
        '127.0.0.1': {
            'db_engine': 'mysql',
            'cdr_type': 'asterisk',
            'db_name': 'asteriskcdrdb',
            'table_name': 'cdr',
            'host': 'localhost',
            'port': '',
            'user': 'root',
            'password': 'password',
        },
        #'192.168.1.200': {
            #'db_engine': 'mysql',
            #'cdr_type': 'asterisk',
            #'db_name': 'asteriskcdrdb',
            #'table_name': 'cdr',
            #'host': 'localhost',
            #'port': '',
            #'user': 'root',
            #'password': 'password',
        #},
    }

To add a new remote Asterisk MySQL CDR store,  ensure that there is a connection to the remote MySQL database, then uncomment the new server settings by removing the # and configuring the credentials to connect to the remote Asterisk CDR store.



.. _realtime-configuration-asterisk:

Realtime configuration for Asterisk
------------------------------------

The Asterisk Manager settings allow CDR-Stats to retrieve Realtime information.

The settings to configure are::

    #Asterisk Manager / Used for Realtime and Concurrent calls
    ASTERISK_MANAGER_HOST = 'localhost'
    ASTERISK_MANAGER_USER = 'cdrstats_user'
    ASTERISK_MANAGER_SECRET = 'cdrstats_secret'


In Asterisk, add a new user in manager.conf, or one of its #include's for CDR-Stats. Further information about Asterisk Manager can be found here : http://www.voip-info.org/wiki/view/Asterisk+config+manager.conf

.. _configuration-freeswitch:

Import configuration for FreeSWITCH
------------------------------------

Freeswitch settings are under the CDR_BACKEND section, and should look as follows::

    CDR_BACKEND = {
        '127.0.0.1': {
            'db_engine': 'mongodb',  # mysql, pgsql, mongodb
            'cdr_type': 'freeswitch',  # asterisk or freeswitch
            'db_name': 'freeswitch_cdr',
            'table_name': 'cdr',  # collection if mongodb
            'host': 'localhost',
            'port': 3306,  # 3306 mysql, 5432 pgsql, 27017 mongodb
            'user': '',
            'password': '',
        },
        #'192.168.1.15': {
        #    'db_engine': 'mongodb',  # mysql, pgsql, mongodb
        #    'cdr_type': 'freeswitch',  # asterisk or freeswitch
        #    'db_name': 'freeswitch_cdr',
        #    'table_name': 'cdr',  # collection if mongodb
        #    'host': 'localhost',
        #    'port': 3306,  # 3306 mysql, 5432 pgsql, 27017 mongodb
        #    'user': '',
        #    'password': '',
        #},
    }


To connect a new Freeswitch system to CDR-Stats, ensure that port 27017 TCP is ONLY open to
the CDR-Stats server on the remote system, then uncomment the settings by removing the #,
and configure the IP address and db_name to match those in the mod_cdr_mongodb configuration
as described at :
http://www.cdr-stats.org/documentation/beginners-guide/howto-installing-on-freeswitch/

Configuring Email
-----------------

To configure the SMTP client so that reports and alerts are sent via email, edit 
/usr/share/cdr-stats/settings_local.py, and identify the email section.

#EMAIL BACKEND
#=============
# Email configuration
DEFAULT_FROM_EMAIL = 'CDR-Stats <cdr-...@localhost.com>'
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_USE_TLS = True
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_PORT = 587
EMAIL_HOST_USER = 'user...@gmail.com'
EMAIL_HOST_PASSWORD = 'password'
EMAIL_SUBJECT_PREFIX = '[CDR-Stats] '

Fill in the details to match your SMTP server. The above example is for Gmail. When done, restart Celery and Apache.

To test that the email is working, from the command line type:

$workon cdr-stats
$python manage.py send_daily_report




cdr_stats/settings.py
#
# CDR-Stats License
# http://www.cdr-stats.org
#
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this file,
# You can obtain one at http://mozilla.org/MPL/2.0/.
#
# Copyright (C) 2011-2013 Star2Billing S.L.
#
# The Initial Developer of the Original Code is
# Arezqui Belaid <info@star2billing.com>
#
import os
import djcelery
djcelery.setup_loader()


DEBUG = False
TEMPLATE_DEBUG = False

ADMINS = (
    ('Your Name', 'your_email@domain.com'),
)
MANAGERS = ADMINS

SERVER_EMAIL = 'cdr-stats@localhost.com'

APPLICATION_DIR = os.path.dirname(globals()['__file__'])

DATABASES = {
    'default': {
        # 'postgresql_psycopg2','postgresql','sqlite3','oracle', 'django.db.backends.mysql'
        'ENGINE': 'django.db.backends.sqlite3',
        # Database name or path to database file if using sqlite3.
        'NAME': APPLICATION_DIR + '/database/cdr_stats.db',
        'USER': '',                      # Not used with sqlite3.
        'PASSWORD': '',                  # Not used with sqlite3.
        'HOST': '',                      # Not used with sqlite3.
        'PORT': '',                      # Not used with sqlite3.
    }
}

CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
        'LOCATION': '/tmp/',
    }
}

#Include for cache machine : http://jbalogh.me/projects/cache-machine/
CACHE_BACKEND = 'caching.backends.locmem://'

#Calls to QuerySet.count() can be cached,
CACHE_COUNT_TIMEOUT = 60  # seconds, not too long.

# Local time zone for this installation. Choices can be found here:
# http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
# although not all choices may be available on all operating systems.
# On Unix systems, a value of None will cause Django to use the same
# timezone as the operating system.
# If running in a Windows environment this must be set to the same as your
# system time zone.
TIME_ZONE = 'America/Chicago'

# Language code for this installation. All choices can be found here:
# http://www.i18nguy.com/unicode/language-identifiers.html
LANGUAGE_CODE = 'en-us'

SITE_ID = 1

# If you set this to False, Django will make some optimizations so as not
# to load the internationalization machinery.
USE_I18N = True

# If you set this to False, Django will not format dates, numbers and
# calendars according to the current locale
USE_L10N = True


# Absolute path to the directory static files should be collected to.
# Don't put anything in this directory yourself; store your static files
# in apps' "static/" subdirectories and in STATICFILES_DIRS.
# Example: "/home/media/media.lawrence.com/static/"
STATIC_ROOT = APPLICATION_DIR + "/static/"

# URL prefix for static files.
# Example: "http://media.lawrence.com/static/"
STATIC_URL = '/static/'

# Absolute path to the directory that holds media.
# Example: "/home/media/media.lawrence.com/"
MEDIA_ROOT = os.path.join(APPLICATION_DIR, 'static')

COUNTRIES_FLAG_PATH = 'flags/%s.png'

# Additional locations of static files
STATICFILES_DIRS = (
    # Put strings here, like "/home/html/static" or "C:/www/django/static".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
    # os.path.join(APPLICATION_DIR, "resources"),
    ("cdr-stats", os.path.join(APPLICATION_DIR, "resources")),
)

# List of finder classes that know how to find static files in
# various locations.
STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
    'dajaxice.finders.DajaxiceFinder',
)

# Make this unique, and don't share it with anybody.
SECRET_KEY = ')ey%^d=pk^jxgam92tdqb0z+0bbhk=7dub_0$ttw#u8yj)rgo$'

# List of callables that know how to import templates from various sources.
TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.Loader',
    'django.template.loaders.app_directories.Loader',
)

MIDDLEWARE_CLASSES = (
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.locale.LocaleMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    #'geordi.VisorMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'linaro_django_pagination.middleware.PaginationMiddleware',
    'common.filter_persist_middleware.FilterPersistMiddleware',
    'mongodb_connection_middleware.MongodbConnectionMiddleware',
)

TEMPLATE_CONTEXT_PROCESSORS = (
    "django.contrib.auth.context_processors.auth",
    "django.core.context_processors.debug",
    "django.core.context_processors.i18n",
    "django.core.context_processors.media",
    "django.core.context_processors.static",
    "django.core.context_processors.csrf",
    "django.contrib.messages.context_processors.messages",
    "django.core.context_processors.request",
)

ROOT_URLCONF = 'cdr_stats.urls'

TEMPLATE_DIRS = (
    # Put strings here, like "/home/html/django_templates"
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
    os.path.join(APPLICATION_DIR, 'templates'),
)

INTERNAL_IPS = ('127.0.0.1')

DEBUG_TOOLBAR_CONFIG = {
    'INTERCEPT_REDIRECTS': False,
}

DAJAXICE_MEDIA_PREFIX = "dajaxice"
#DAJAXICE_MEDIA_PREFIX = "dajax"  # http://domain.com/dajax/
#DAJAXICE_CACHE_CONTROL = 10 * 24 * 60 * 60

INSTALLED_APPS = (
    #admin tool apps
    'admin_tools',
    'admin_tools.theming',
    'admin_tools.menu',
    'admin_tools.dashboard',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.admin',
    'django.contrib.markup',
    'api',
    'cdr',
    'cdr_alert',
    'user_profile',
    'frontend',    
    'dateutil',
    'south',
    'linaro_django_pagination',
    'djcelery',
    'tastypie',
    'django_socketio',
    'common',
    'notification',
    'country_dialcode',
    #'geordi',
    'gunicorn',
    'apiplayground',
    'frontend_notification',
    'dajaxice',
    'dajax',
    'voip_gateway',
    'voip_billing',    
)

# Debug Toolbar
try:
    import debug_toolbar
except ImportError:
    pass
else:
    INSTALLED_APPS = INSTALLED_APPS + ('debug_toolbar',)
    MIDDLEWARE_CLASSES = MIDDLEWARE_CLASSES + \
        ('debug_toolbar.middleware.DebugToolbarMiddleware',)
    DEBUG_TOOLBAR_CONFIG = {
        'INTERCEPT_REDIRECTS': False,
    }

# Nose
try:
    import nose
except ImportError:
    pass
else:
    INSTALLED_APPS = INSTALLED_APPS + ('django_nose',)
    TEST_RUNNER = 'utils.test_runner.MyRunner'

# Debug Toolbar mongo

# commented cause this module doesn't work at the moment
# https://groups.google.com/forum/?fromgroups#!topic/mongoengine-users/cwIdHSNPCwY

try:
    import debug_toolbar_mongo
except ImportError:
    pass
else:
    INSTALLED_APPS = INSTALLED_APPS + ('debug_toolbar_mongo',)

    DEBUG_TOOLBAR_MONGO_STACKTRACES = True
    DEBUG_TOOLBAR_PANELS = (
        'debug_toolbar.panels.version.VersionDebugPanel',
        'debug_toolbar.panels.timer.TimerDebugPanel',
        #Warning: If you run profiling this will duplicate the view execution
        #'debug_toolbar.panels.profiling.ProfilingDebugPanel',
        'debug_toolbar.panels.settings_vars.SettingsVarsDebugPanel',
        'debug_toolbar.panels.headers.HeaderDebugPanel',
        'debug_toolbar.panels.request_vars.RequestVarsDebugPanel',
        'debug_toolbar.panels.template.TemplateDebugPanel',
        'debug_toolbar.panels.sql.SQLDebugPanel',
        'debug_toolbar.panels.signals.SignalDebugPanel',
        'debug_toolbar.panels.logger.LoggingPanel',
    )
    DEBUG_TOOLBAR_PANELS = DEBUG_TOOLBAR_PANELS + \
        ('debug_toolbar_mongo.panel.MongoDebugPanel',)


# Django extensions
try:
    import django_extensions
except ImportError:
    pass
else:
    INSTALLED_APPS = INSTALLED_APPS + ('django_extensions',)


AUTH_PROFILE_MODULE = 'user_profile.UserProfile'

LOG_COLORSQL_ENABLE = True
LOG_COLORSQL_VERBOSE = True

LOGIN_URL = '/pleaselog/'

#DILLA SETTINGS
#==============
DICTIONARY = "/usr/share/dict/words"
DILLA_USE_LOREM_IPSUM = False  # set to True ignores dictionary
DILLA_APPS = ['cdr']
DILLA_SPAMLIBS = ['cdr.cdr_custom_spamlib']
# To use Dilla
# > python manage.py run_dilla --cycles=100


gettext = lambda s: s

LANGUAGES = (
    ('en', gettext('English')),
    ('fr', gettext('French')),
    ('es', gettext('Spanish')),
    ('pt', gettext('Portuguese')),
    ('de', gettext('German')),
    ('ru', gettext('Russian')),
)

# News URL
NEWS_URL = 'http://www.cdr-stats.org/news.php'

#DJANGO-ADMIN-TOOL
#=================
ADMIN_TOOLS_MENU = 'cdr_stats.custom_admin_tools.menu.CustomMenu'
ADMIN_TOOLS_INDEX_DASHBOARD = \
    'cdr_stats.custom_admin_tools.dashboard.CustomIndexDashboard'
ADMIN_TOOLS_APP_INDEX_DASHBOARD = \
    'cdr_stats.custom_admin_tools.dashboard.CustomAppIndexDashboard'
ADMIN_MEDIA_PREFIX = '/static/admin/'

#CELERY
#======
CARROT_BACKEND = 'redis'

BROKER_HOST = 'localhost'  # Maps to redis host
BROKER_PORT = 6379         # Maps to redis port
BROKER_VHOST = 0           # Maps to database number

CELERY_RESULT_BACKEND = 'redis'
REDIS_HOST = 'localhost'
REDIS_PORT = 6379
REDIS_DB = 0
#REDIS_CONNECT_RETRY = True

CELERY_DEFAULT_QUEUE = 'cdrstats'
CELERY_DEFAULT_EXCHANGE = "cdrstats_tasks"
CELERY_DEFAULT_EXCHANGE_TYPE = "topic"
CELERY_DEFAULT_ROUTING_KEY = "task.cdrstats"
CELERY_QUEUES = {
    'cdrstats': {
        'binding_key': '#',
    },
}

#SOCKETIO
#========
SOCKETIO_HOST = 'localhost'
SOCKETIO_PORT = 9000
SOCKETIO_CALLNUM_DEFAULT = 0

#GENERAL
#=======
# PREFIX_LIMIT_MIN & PREFIX_LIMIT_MAX are used to know
# how many digits are used to match against the dialcode prefix database
PREFIX_LIMIT_MIN = 2
PREFIX_LIMIT_MAX = 5

# If PN is lower than PN_MIN_DIGITS it will be considered as an extension
# If PN is longer than PN_MIN_DIGITS but lower than PN_MAX_DIGITS then
# The PN will be considered as local call and the LOCAL_DIALCODE will be added
LOCAL_DIALCODE = 1  # Set the Dialcode of your country (44 for UK, 1 for US)
PN_MIN_DIGITS = 6
PN_MAX_DIGITS = 9

# List of phonenumber prefix to ignore, this will be remove prior analysis
PREFIX_TO_IGNORE = "+,0,00,000,0000,00000,011,55555,99999"

# When the dialed number is less or equal to INTERNAL_CALL, the call will be considered
# as a internal call, for example when dialed number is 41200
INTERNAL_CALL = 5

#Realtime Graph : set the Y axis limit
REALTIME_Y_AXIS_LIMIT = 300

#ASTERISK IMPORT
#===============
ASTERISK_PRIMARY_KEY = 'acctid'  # acctid, _id

#CDR_BACKEND
#===========
#list of CDR Backends to import
CDR_BACKEND = {
    # '127.0.0.1': {
    #     'db_engine': 'mysql',  # mysql, pgsql, mongodb
    #     'cdr_type': 'asterisk',  # asterisk or freeswitch
    #     'db_name': 'asteriskcdr',
    #     'table_name': 'cdr',  # collection if mongodb
    #     'host': 'localhost',
    #     'port': 3306,  # 3306 mysql, 5432 pgsql, 27017 mongodb
    #     'user': 'root',
    #     'password': 'password',
    # },
    '127.0.0.1': {
        'db_engine': 'mongodb',  # mysql, pgsql, mongodb
        'cdr_type': 'freeswitch',  # asterisk or freeswitch
        'db_name': 'freeswitch_cdr',
        'table_name': 'cdr',  # collection if mongodb
        'host': 'localhost',
        'port': 27017,  # 3306 mysql, 5432 pgsql, 27017 mongodb
        'user': '',
        'password': '',
    },
}

#Define the IP of your local Switch, it needs to exist in the CDR_BACKEND list
LOCAL_SWITCH_IP = '127.0.0.1'

#Asterisk Manager / Used for Realtime and Concurrent calls
ASTERISK_MANAGER_HOST = 'localhost'
ASTERISK_MANAGER_USER = 'cdrstats_user'
ASTERISK_MANAGER_SECRET = 'cdrstats_secret'

#MONGODB
#=======
#Settings of CDR-Stats MongoDB server, this is used to store the analytic data
MONGO_CDRSTATS = {
    'DB_NAME': 'cdr-stats',
    'HOST': 'localhost',
    'PORT': 27017,
    'CDR_COMMON': 'cdr_common',
    'DAILY_ANALYTIC': 'daily_analytic',
    'MONTHLY_ANALYTIC': 'monthly_analytic',
    'CONC_CALL': 'concurrent_call',
    'CONC_CALL_AGG': 'concurrent_call_aggregate'
}

#API PLAYGROUND
#==============
API_PLAYGROUND_FEEDBACK = False

#No of records per page
#=======================
PAGE_SIZE = 10

#TOTAL_GRAPH_COLOR For TOTAL Variable
#====================================
TOTAL_GRAPH_COLOR = '#A61700'

# Display Total Countries
#========================
NUM_COUNTRY = 10


#TASTYPIE API
#============
API_ALLOWED_IP = ['127.0.0.1', 'localhost']

#EMAIL BACKEND
#=============
# Email configuration
DEFAULT_FROM_EMAIL = 'CDR-Stats <cdr-stats@localhost.com>'
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_USE_TLS = True
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_PORT = 587
EMAIL_HOST_USER = 'username@gmail.com'
EMAIL_HOST_PASSWORD = 'password'
EMAIL_SUBJECT_PREFIX = '[CDR-Stats] '
# Use only in Debug mode. Not in production
#EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
#EMAIL_BACKEND = 'django.core.mail.backends.dummy.EmailBackend'

#Define the delay in minute between mail notification
#This setting avoid getting span with loads of alarms
DELAY_BETWEEN_MAIL_NOTIFICATION = 10

#IPYTHON
#=======
IPYTHON_ARGUMENTS = [
    '--ext', 'django_extensions.management.notebook_extension',
    '--profile=nbserver',
    '--debug'
]

#IMPORT LOCAL SETTINGS
#=====================
try:
    from settings_local import *
except:
    pass
